SeleniumTests/LoginTutByTests.cs

```diff
@@ -13,27 +16,69 @@ public class LoginTutByTests
         public void StartBrowser()
         {
             driver = new ChromeDriver(@"C:\Users\aleks\Downloads");
+            driver.Manage().Timeouts().ImplicitWait = TimeSpan.FromSeconds(20);
             driver.Url = "http://tut.by";
         }
 
         [Test]
-        public void LoginTest()
+        [TestCase("seleniumtests@tut.by", "123456789zxcvbn")]
+        [TestCase("seleniumtests2@tut.by", "123456789zxcvbn")]
+        public void LoginTest(string login, string password)
         {
             var findMethods = new FindMethods();
             findMethods.FindByXPath(driver, "//a[@class = 'enter']").Click();
             var loginInput = driver.FindElement(By.XPath("//input[@name = 'login']"));
-            loginInput.SendKeys("aegorova@tut.by");
+            loginInput.SendKeys(login);
 
             var passwordInput = driver.FindElement(By.XPath("//input[@name = 'password']"));
-            passwordInput.SendKeys("mtfbwy8772902");
+            passwordInput.SendKeys(password);
 
             var submitButton = driver.FindElement(By.XPath("//input[@type = 'submit']"));
             submitButton.SendKeys(Keys.Enter);
 
+            Thread.Sleep(2000); //explicit wait
+
             var logedInSection = driver.FindElement(By.XPath("//a[contains(@class,'logedin')]"));
             Assert.True(logedInSection.Displayed,"Success element is not displayed");
         }
 
+        [Test]
+        [TestCase("seleniumtests@tut.by", "123456789zxcvbn")]
+        [TestCase("seleniumtests2@tut.by", "123456789zxcvbn")]
+        public void LoginTestWithExplicitWaiter(string login, string password)
+        {
+            driver.FindElement(By.XPath("//a[@class = 'enter']")).Click();
+            var loginInput = driver.FindElement(By.XPath("//input[@name = 'login']"));
+            loginInput.SendKeys(login);
+
+            var passwordInput = driver.FindElement(By.XPath("//input[@name = 'password']"));
+            passwordInput.SendKeys(password);
+
+            var submitButton = driver.FindElement(By.XPath("//input[@type = 'submit']"));
+            submitButton.SendKeys(Keys.Enter);
+
+            var wait = new WebDriverWait(driver, TimeSpan.FromSeconds(15));
+
+            var userName = wait.Until(condition =>
+            {
+                try
+                {
+                    var elementToBeDisplayed = driver.FindElement(By.XPath("//span[@class='uname']"));
+                    return elementToBeDisplayed.Displayed;
+                }
+                catch (StaleElementReferenceException)
+                {
+                    return false;
+                }
+                catch (NoSuchElementException)
+                {
+                    return false;
+                }
+            });
```

Comment:
This wait method have to excluded as a separate method. Make it universe to use for any web elements. This method should have signature like this WaitUntilDisplayed(By locator, TimeSpan time).